.github/ci.sh

```diff
@@ -10,7 +10,7 @@ mkdir -p "$BIN"
 is_exe() { [[ -x "$1/$2$EXT" ]] || command -v "$2" > /dev/null 2>&1; }
 
 extract_exe() {
-  exe="$(cabal v2-exec which "$1$EXT")"
+  exe="$(cabal v2-exec which "$1$EXT" | tail -1)"
```

Comment:
I'm starting to think that using find directly on the dist-newstyle folder might achieve better results. It seems to be a more common practice than `cabal v2-exec` from other repos I've seen as well.

Reply:
Yeah, that might be the way to go. I've had endless trouble with `v2-exec`. I'll update it.

Reply:
@atomb, I'm curious as to what problems you saw with `v2-exec`.  The advantage of the `v2-exec` method over the `find` method is that the `dist-newstyle` can have multiple instances of the binary because it creates different directories based on architecture, GHC version, etc., so the find might not get the one for the current configuration (esp. with caching of builds).

Reply:
One some platforms, for reasons I don't entirely understand, the output of `v2-exec which ...` includes extra text, and it can be hard to extract what you need. Usually, `tail` is enough, but it seems like more of a hack. I'm hoping that in CI the issue of multiple instances of a particular binary won't show up, but maybe I'm wrong about that.

Reply:
@jared-w, is the github actions caching a potential issue for the `find`-based approach returning the wrong executable?